producer.go

```diff
@@ -45,6 +45,8 @@ func NewProducer(cfg *ProducerConfig, interceptors ...ProducerInterceptor) (Prod
 		AllowAutoTopicCreation: cfg.Writer.AllowAutoTopicCreation,
 	}
 
+	cfg.Logger = NewZapLogger(cfg.LogLevel)
```

Comment:
Instead of opening a new logger and defining Logger and LogLevel in ProducerConfig, couldn't we use it from the base? I think this should not be necessary.